addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1104,10 +1104,14 @@ const Wysiwyg = Widget.extend({
                     });
                     colorpicker.on('custom_color_picked color_picked', null, ev => {
                         this._processAndApplyColor(eventName, ev.data.color);
-                        this.odooEditor.historyStep();
                     });
                     colorpicker.on('color_hover color_leave', null, ev => {
-                        this._processAndApplyColor(eventName, ev.data.color);
+                        this.odooEditor.historyPauseSteps();
+                        try {
+                            this._processAndApplyColor(eventName, ev.data.color);
+                        } finally {
+                            this.odooEditor.historyUnpauseSteps();
+                        }
```

Comment:
I would use `odooEditor.observerActive` and `odooEditor.observerUnactive` rather than `historyPauseSteps` and `historyUnpauseSteps`, otherwise, all the mutations will be saved rather than skipped.

Reply:
Is not the point exactly to save the mutations rather than skip them? :thinking:

Reply:
Oh no, spoke to fast. Is there a case where historyPauseSteps is needed then? :thinking:

Reply:
No history step is saved at all if I do this (or without the label)
(i.e. ctrl+z removes the dropped snippet instead of cancelling color selection)
```js
                        this.odooEditor.observerUnactive("colorpreview");
                        try {
                            this._processAndApplyColor(eventName, ev.data.color);
                        } finally {
                            this.odooEditor.observerActive("colorpreview");
                        }
```
even if I explicitly restore the `this.odooEditor.historyStep();` on the apply part.

Reply:
The reason for skipping the mutations is for performance. Otherwise it generate a lot of unnecessary data that need to be processed and send over the network.

The difference of `historyPauseSteps` is to continue to aggregate mutations into one step rather than multiples steps. It could be used to execute commands that will be saved as one step.

I've tried with `odooEditor.observerActive` and `odooEditor.observerUnactive` unsuccessfully also and I cannot easily provide the reason why.

So finally, I think it is acceptable the current commit as is.
We will probably make a "history compression" later and therefore will take care of the performance overhead with that commit.